src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyExecutionManagerWithDataCollection.cs

```diff
@@ -142,32 +138,33 @@ public override int StartTestRun(TestRunCriteria testRunCriteria, ITestRunEvents
         /// <inheritdoc/>
         public override void Cancel()
         {
-            this.ProxyDataCollectionManager?.AfterTestRunEnd(isCanceled: true, runEventsHandler: this.DataCollectionRunEventsHandler);
+            try
+            {
+                this.ProxyDataCollectionManager.AfterTestRunEnd(isCanceled: true, runEventsHandler: this.DataCollectionRunEventsHandler);
+            }
+            catch (Exception ex)
```

Comment:
Suggest to just log the exception in `finally` and remove `catch`. Don't catch `Exception` to throw again another `Exception`.